### PR TITLE
Add path find response field full_reply (RIPD-894)

### PR DESCRIPTION
**Pull request by miguelportilla**
The new JSON field "full_reply" is true if it's a full-depth search and false if it's a partial/quick search.

Reviewers: @ximinez @rec

### CODE REVIEW

**Comment by rec** [comment]
:+1: Obviously correct.

**Comment by ximinez** [comment]
Can we get some unit / integration tests for this new field?

**Comment by rec** [comment]
Oops, yes.  Good question!

**Comment by vinniefalco** [comment]
Are you shitting me? We don't need tests for this!! ARGH

**Comment by ximinez** [comment]
> Are you shitting me? We don't need tests for this!! ARGH

Ok, that's fine, but here's my reasoning: At my previous job, on several projects, we had woefully inadequate test coverage. We put an informal rule in place that any new code, no matter how trivial, had to be accompanied by tests. It was slow and painful at first, because to test the new code, you generally had to test the old code, too. But, that increased our code coverage. Eventually some areas of the code got to the point where they were covered well enough that a trivial change to the functionality only needed to be accompanied by a trivial change in the tests, but we still insisted on updating the tests to ensure that they didn't get stale.

*tl;dr*: If you don't have test coverage, trivial changes are a great opportunity to add that test coverage. If you **do** have test coverage, trivial changes only require trivial updates to the tests. Stale tests are not good.

**Comment by vinniefalco** [comment]
There HAS to be a line on one side of which we write tests and the other side we don't. "Always write tests" makes no sense at all, there is such a thing as too much test coverage. For example, multisigning does NOT have to test each possible transaction type or combination of payment options. Of course we would all love programming to be so easy that it could be reduced to black and white rules like "always write tests" but that is simply not reality.

**Comment by rec** [comment]
Unit test code coverage is a Ripple Labs OKR.  

We shouldn't waste time arguing over whether code deserves a unit test - we should just write it as a matter of course.  I've been in projects that had close to 100% unit test coverage, and the code quality was amazingly good as a result.

**Comment by vinniefalco** [comment]
No we should not
